src/main/java/ch/njol/skript/conditions/CondIsWithinLocation.java

```diff
@@ -0,0 +1,79 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.AABB;
+import ch.njol.util.Kleenean;
+import org.bukkit.Location;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Is Within Location")
+@Description({
+	"Whether a location is within two other locations forming a cuboid.",
+	"Using the <a href='conditions.html#CondCompare'>is between</a> condition will refer to a straight line between locations."
+})
+@Examples({
+	"if player's location is within {_loc1} and {_loc2}:",
+		"\tsend \"You are in a PvP zone!\" to player"
+})
+@Since("INSERT VERSION")
+public class CondIsWithinLocation extends Condition {
+
+	static {
+		PropertyCondition.register(CondIsWithinLocation.class, "within %location% and %location%", "locations");
+	}
+
+	private Expression<Location> locsToCheck, loc1, loc2;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setNegated(matchedPattern == 1);
+		locsToCheck = (Expression<Location>) exprs[0];
+		loc1 = (Expression<Location>) exprs[1];
+		loc2 = (Expression<Location>) exprs[2];
+		return true;
+	}
+
+	@Override
+	public boolean check(Event event) {
+		Location one = loc1.getSingle(event);
+		Location two = loc2.getSingle(event);
+		if (one == null || two == null || one.getWorld() != two.getWorld())
```

Comment:
you *might* need to do a check that `one.getWorld()` is not null

Reply:
you don't need to check if a value is null before comparing, it does that for you

Reply:
I did not know if AABB was weird about locations will a null world

ex: if `one` and `two` are both locations without worlds, will AABB be problematic :D

Reply:
if `one` and `two` both don't have worlds then the location its checking also has to have no world, otherwise it'll return false. So, if it'll only return true if none of the locations have worlds